addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -457,6 +463,19 @@ export function createRelatedModels(modelDefs, modelClasses = {}, opts = {}) {
         const Model = modelClasses[model] || Base;
         if (!(model in setTrapsCache)) {
             setTrapsCache[model] = function setTrap(target, prop, value, receiver) {
+                if (["date", "datetime"].includes(target.model?.modelFields?.[prop]?.type)) {
+                    // Verify if is already a valid dateobject
+                    if (!(value instanceof Object) && value) {
+                        const datetime = deserializeDateTime(value);
+                        if (!datetime.isValid) {
+                            throw new Error(
+                                `Invalid date: ${value} for model ${target.model.modelName} in field ${prop}`
+                            );
+                        }
+                        value = datetime;
+                    }
+                }
+
```

Comment:
I think overloading the set trap should be the last resort. You can still put this logic in the "update" method.